src/components/study-container.js

```diff
@@ -0,0 +1,455 @@
+import * as PropTypes from 'prop-types';
+import { useParams } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+    connectNotificationsWebsocket,
+    fetchAllEquipments,
+    fetchLoadFlowInfos,
+    fetchNetworkModificationTree,
+    fetchSecurityAnalysisStatus,
+    fetchStudyExists,
+} from '../utils/rest-api';
+import {
+    closeStudy,
+    filteredNominalVoltagesUpdated,
+    loadNetworkModificationTreeSuccess,
+    networkCreated,
+    openStudy,
+    selectTreeNode,
+    studyUpdated,
+} from '../redux/actions';
+import Network from './network/network';
+import { equipments } from './network/network-equipments';
+import WaitingLoader from './util/waiting-loader';
+import { displayErrorMessageWithSnackbar, useIntlRef } from '../utils/messages';
+import NetworkModificationTreeModel from './graph/network-modification-tree-model';
+import { useSnackbar } from 'notistack';
+import {
+    getSecurityAnalysisRunningStatus,
+    RunningStatus,
+} from './util/running-status';
+import { useIntl } from 'react-intl';
+
+export function useNodeData(
+    studyUuid,
+    selectedNodeUuid,
+    fetcher,
+    invalidations,
+    defaultValue,
+    resultConversion
+) {
+    const [result, setResult] = useState(defaultValue);
+    const [isPending, setIsPending] = useState(false);
+    const [errorMessage, setErrorMessage] = useState(undefined);
+    const selectedNodeUuidRef = useRef();
+    const studyUpdatedForce = useSelector((state) => state.studyUpdated);
+    const lastUpdateRef = useRef();
+
+    const update = useCallback(() => {
+        selectedNodeUuidRef.current = selectedNodeUuid;
+        setIsPending(true);
+        fetcher(studyUuid, selectedNodeUuid)
+            .then((res) => {
+                if (selectedNodeUuidRef.current === selectedNodeUuid)
+                    setResult(resultConversion ? resultConversion(res) : res);
+            })
+            .catch((err) => setErrorMessage(err.message))
+            .finally(() => setIsPending(false));
+    }, [
+        selectedNodeUuid,
+        studyUuid,
+        setResult,
+        setErrorMessage,
+        setIsPending,
+        selectedNodeUuidRef,
+        fetcher,
+        resultConversion,
+    ]);
+
+    /* initial fetch and update */
+    useEffect(() => {
+        if (!studyUuid || !selectedNodeUuid) return;
+        const headers = studyUpdatedForce?.eventData?.headers;
+        const updateType = headers && headers['updateType'];
+        const node = headers && headers['node'];
+        const isUpdateForUs =
+            lastUpdateRef.current !== studyUpdatedForce &&
+            updateType &&
+            node === selectedNodeUuid &&
+            invalidations.find((e) => updateType === e) !== -1;
+        lastUpdateRef.current = studyUpdatedForce;
+        if (selectedNodeUuidRef.current !== selectedNodeUuid || isUpdateForUs) {
+            update();
+        }
+    }, [
+        update,
+        selectedNodeUuidRef,
```

Comment:
ref couldn't be dependencies because they never change.

Reply:
done